modules/multi-architecture-modify-machine-set-gcp.adoc

```diff
@@ -0,0 +1,140 @@
+----
+$ oc create -f gcp-arm64-machine-set-0.yaml
+----
++
+--
+.Sample GCP YAML compute machine set to deploy an ARM64 compute node
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_id> <1>
+  name: <infrastructure_id>-w-a
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_id>-w-a
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+        machine.openshift.io/cluster-api-machine-role: <role> <2>
+        machine.openshift.io/cluster-api-machine-type: <role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_id>-w-a
+    spec:
+      metadata:
+        labels:
+          node-role.kubernetes.io/<role>: ""
+      providerSpec:
+        value:
+          apiVersion: gcpprovider.openshift.io/v1beta1
+          canIPForward: false
+          credentialsSecret:
+            name: gcp-cloud-credentials
+          deletionProtection: false
+          disks:
+          - autoDelete: true
+            boot: true
+            image: <path_to_image> <3>
+            labels: null
+            sizeGb: 128
+            type: pd-ssd
+          gcpMetadata: <4>
+          - key: <custom_metadata_key>
+            value: <custom_metadata_value>
+          kind: GCPMachineProviderSpec
+          machineType: n1-standard-4 <5>
+          metadata:
+            creationTimestamp: null
+          networkInterfaces:
+          - network: <infrastructure_id>-network
+            subnetwork: <infrastructure_id>-worker-subnet
+          projectID: <project_name> <6>
+          region: us-central1 <7>
+          serviceAccounts:
+          - email: <infrastructure_id>-w@<project_name>.iam.gserviceaccount.com
+            scopes:
+            - https://www.googleapis.com/auth/cloud-platform
+          tags:
+            - <infrastructure_id>-worker
+          userDataSecret:
+            name: worker-user-data
+          zone: us-central1-a
+----
+<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the OpenShift CLI installed, you can obtain the infrastructure ID by running the following command:
++
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+<2> Specify the role node label to add.
+<3> Specify the path to the image that is used in current compute machine sets. You can use the following commands to access a ARM64 supported image. 
++
+The path to the image should follow the following format:
++
+[source,terminal]
+----
+projects/${project}/global/images/<image_name>
+----
+To access the image name, run the following command: 
```

Comment:
My apologies for my confusion on this one, its just these two items that are needed right? @aleskandro

Reply:
Let me show you... The data we are retrieving with the command below comes from [this json](https://github.com/openshift/installer/blob/master/data/data/coreos/rhcos.json#L235-L239).
```json
"architectures": {
   "aarch64": {
      "images": {
        "gcp": {
          "release": "415.92.202309142014-0",
          "project": "rhcos-cloud",
          "name": "rhcos-415-92-202309142014-0-gcp-aarch64"
        }
  ......

```

users must set the path to the image as in row 103. From the example above, it would be :

`projects/rhcos-cloud/global/images/rhcos-415-92-202309142014-0-gcp-aarch64`

I think the options we have are two:

#### Document two similar commands to retrieve (i) the project name, (ii) the image name:

project name:
```bash
oc get configmap/coreos-bootimages \
  -n openshift-machine-config-operator \
  -o jsonpath='{.data.stream}' | jq \
  -r '.architectures.aarch64.images.gcp.project'

[example output] 
rhcos-cloud
```

image name:

```bash
oc get configmap/coreos-bootimages \
  -n openshift-machine-config-operator \
  -o jsonpath='{.data.stream}' | jq \
  -r '.architectures.aarch64.images.gcp.name'

[example output]  
rhcos-415-92-202309142014-0-gcp-aarch64
```

#### Document only one command

we can also let the users print the json and specify that they need to use the `project` and `name` fields to fill the path ` projects/<project>/global/images/<name>` based on the json output of the command:

```bash
oc get configmap/coreos-bootimages \
  -n openshift-machine-config-operator \
  -o jsonpath='{.data.stream}' | jq \
  -r '.architectures.aarch64.images.gcp'

[example output]  
        "gcp": {
          "release": "415.92.202309142014-0",
          "project": "rhcos-cloud",
          "name": "rhcos-415-92-202309142014-0-gcp-aarch64"
        }
```

The first way gives users two redundant commands that provide specific output for each missing part of the path URI. The second way gives one only command and requires the users to look at the json output to fille the image path.

What do you think is the best to provide?

Reply:
That actually helps a bunch, thank you Alessandro. I like having just the one output and I can do some rewording so that users know what part they need.

Reply:
@aleskandro Hello! I just updated the PR and I'm more confident about the updates, WDYT?

Reply:
i think the example output above should also be included mentioning that the `project` and `name` field should be used respectively (as Alessandro already mentioned):

```
[example output]  
        "gcp": {
          "release": "415.92.202309142014-0",
          "project": "rhcos-cloud",
          "name": "rhcos-415-92-202309142014-0-gcp-aarch64"
        }
```

Reply:
+1 to @Prashanth684

Reply:
Just added the example output and did some rewording! Thank you @Prashanth684 and @aleskandro, should be all set now